### ISSUE TITLE
JForm B/C broken

### ISSUE THREAD

**Issue by phproberto** [joomla/joomla-cms#15977]
### Steps to reproduce the issue

Having a form defined like this:

```xml
<?xml version="1.0" encoding="utf-8"?>
<form>
    <fields name="custom_fields" />
</form>
```

Note: I've only added the `fields` section to explain how this issue happens. `custom_fields` is just an empty fields section there to allow field injection.

Before you could do this to add a field to a form `fields` section:

```php
		$xml = '<field type="text" name="test-field" label="Test field" />';

		$form->setField(new \simpleXmlElement($xml), 'custom_fields');
```

And retrieve the field like:

`$field = $form->getField('test-field', 'custom_fields');`

Which was returning the correct field.

After this:
https://github.com/joomla/joomla-cms/pull/14737

It looks like we are now forcing that a fieldset exists on the `fields` section:

`//fields[@name="custom_fields"]/fieldset[@name="default"]`

because the new `$fieldset` param added to the function:
https://github.com/joomla/joomla-cms/pull/14737/files#diff-ed54af5d39d21ed9a84d28eba903bd5fR1082

So the field is never return.

### Expected result

I'd expect B/C. So if fieldset was not required before now it shouldn't. Before this XML worked:

```xml
<fields name="custom_fields">
        <!-- fields here -->
</fields>
```

### Actual result

Now fieldset is mandatory so you need an structure like:

```xml
<fields name="custom_fields">
    <fieldset name="default">
        <!-- fields here -->
    </fieldset>
</fields>
```

### System information (as much as possible)

Joomla v3.7.0

### Additional comments

See https://github.com/joomla/joomla-cms/pull/14737

**Comment by phproberto** [comment]
No. I didn't test `v3.7.0`. I know I should be banned from the project for not doing so and everything since Windows Me is my fault. Let's skip that part. TY.

**Comment by Bakual** [comment]
@rdeutz since that was your PR, can you have a look?

**Comment by rdeutz** [comment]
@Bakual will have a look but first I need to process the ban for roberto :-)

**Comment by phproberto** [comment]
$fieldset should maybe have a `null` value but probably that removes the sense of adding such parameter?

Ping me when I'm banned so I can start being a hero on twitter 💃

**Comment by phproberto** [comment]
![isformbrokenbyrobertdeutz](https://cloud.githubusercontent.com/assets/1119272/25989775/bbc733d6-36fd-11e7-9da3-c2d3e9875d31.jpg)

**Comment by rdeutz** [comment]
closed because we have  a PR #15981 to test
